packages/wallet-sdk/src/sign/scw/utils.ts

```diff
@@ -220,3 +247,242 @@ export function fillMissingParamsForFetchPermissions(
+    domain: {
+      name: 'Spend Permission Manager',
+      version: '1',
+      chainId: chainId,
+      verifyingContract: spendPermissionManagerAddress,
+    },
+    types: {
+      SpendPermission: [
+        { name: 'account', type: 'address' },
+        { name: 'spender', type: 'address' },
+        { name: 'token', type: 'address' },
+        { name: 'allowance', type: 'uint160' },
+        { name: 'period', type: 'uint48' },
+        { name: 'start', type: 'uint48' },
+        { name: 'end', type: 'uint48' },
+        { name: 'salt', type: 'uint256' },
+        { name: 'extraData', type: 'bytes' },
+      ],
+    },
+    primaryType: 'SpendPermission',
+    message: {
+      account: spendPermission.account,
+      spender: spendPermission.spender,
+      token: spendPermission.token,
+      allowance: spendPermission.allowance,
+      period: spendPermission.period,
+      start: spendPermission.start,
+      end: spendPermission.end,
+      salt: spendPermission.salt,
+      extraData: spendPermission.extraData,
+    },
+  } as const;
+}
+
+export function createSpendPermissionBatchMessage({
+  spendPermissionBatch,
+  chainId,
+}: {
+  spendPermissionBatch: SpendPermissionBatch;
+  chainId: number;
+}) {
+  return {
+    domain: {
+      name: 'Spend Permission Manager',
+      version: '1',
+      chainId,
+      verifyingContract: spendPermissionManagerAddress,
+    },
+    types: {
+      SpendPermissionBatch: [
+        { name: 'account', type: 'address' },
+        { name: 'period', type: 'uint48' },
+        { name: 'start', type: 'uint48' },
+        { name: 'end', type: 'uint48' },
+        { name: 'permissions', type: 'PermissionDetails[]' },
+      ],
+      PermissionDetails: [
+        { name: 'spender', type: 'address' },
+        { name: 'token', type: 'address' },
+        { name: 'allowance', type: 'uint160' },
+        { name: 'salt', type: 'uint256' },
+        { name: 'extraData', type: 'bytes' },
+      ],
+    },
+    primaryType: 'SpendPermissionBatch',
+    message: {
+      account: spendPermissionBatch.account,
+      period: spendPermissionBatch.period,
+      start: spendPermissionBatch.start,
+      end: spendPermissionBatch.end,
+      permissions: spendPermissionBatch.permissions.map((p) => ({
+        spender: p.spender,
+        token: p.token,
+        allowance: p.allowance,
+        salt: p.salt,
+        extraData: p.extraData,
+      })),
+    },
+  } as const;
+}
+
+export async function waitForCallsTransactionHash({
+  client,
+  id,
+}: { client: PublicClient; id: string }) {
+  const result = await waitForCallsStatus(client, {
+    id,
+  });
+
+  if (result.status === 'success') {
+    return result.receipts?.[0].transactionHash;
+  }
+
+  throw standardErrors.rpc.internal('failed to send transaction');
+}
+
+export function createWalletSendCallsRequest({
+  to,
+  data,
+  value,
+  from,
+  chainId,
+}: {
+  to: Address;
+  data: Hex;
+  value: Hex;
+  from: Address;
+  chainId: Hex;
+}) {
```

Comment:
This has been changed to accept an array of calls instead of a single call in https://github.com/coinbase/coinbase-wallet-sdk/pull/1612